From d35d5af4a10f189df3df2ae44aa6042add660216 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sat, 12 Feb 2022 08:28:29 +0530
Subject: [PATCH] Fix checking Windows platform with preprocessor

Check _WIN32 instead of _MSC_VER because the later one is for MSVC only.
Also __MINGW64__ is defined for 64 bit only. So, chekcing _WIN32 includes
all cases. This fixes building with 32 bit mingw-w64 toolchain.
---
 src/utilities/_hypre_utilities.h | 2 +-
 src/utilities/hopscotch_hash.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/utilities/_hypre_utilities.h b/src/utilities/_hypre_utilities.h
index 5b07583..37a2dff 100644
--- a/src/utilities/_hypre_utilities.h
+++ b/src/utilities/_hypre_utilities.h
@@ -1947,7 +1947,7 @@ static inline HYPRE_Int
 first_lsb_bit_indx( hypre_uint x )
 {
    HYPRE_Int pos;
-#if defined(_MSC_VER) || defined(__MINGW64__)
+#ifdef _WIN32
    if (x == 0)
    {
       pos = 0;
diff --git a/src/utilities/hopscotch_hash.h b/src/utilities/hopscotch_hash.h
index 7ad3888..28f7c3a 100644
--- a/src/utilities/hopscotch_hash.h
+++ b/src/utilities/hopscotch_hash.h
@@ -133,7 +133,7 @@ static inline HYPRE_Int
 first_lsb_bit_indx( hypre_uint x )
 {
    HYPRE_Int pos;
-#if defined(_MSC_VER) || defined(__MINGW64__)
+#ifdef _WIN32
    if (x == 0)
    {
       pos = 0;
-- 
2.35.1

